yarn-project/end-to-end/src/e2e_event_logs.test.ts

```diff
@@ -110,5 +119,46 @@ describe('Logs', () => {
       const badEvent1 = TestLogContract.events.ExampleEvent0.decode(decryptedLog1!.payload);
       expect(badEvent1).toBe(undefined);
     });
+
+    it('emits multiple events as encrypted logs and decodes them', async () => {
+      const randomness = makeTuple(5, makeTuple.bind(undefined, 2, Fr.random)) as Tuple<Tuple<Fr, 2>, 5>;
+      const preimage = makeTuple(5, makeTuple.bind(undefined, 4, Fr.random)) as Tuple<Tuple<Fr, 4>, 5>;
+
+      let i = 0;
+      const firstTx = await testLogContract.methods.emit_encrypted_events(randomness[i], preimage[i]).send().wait();
+      await Promise.all(
+        [...new Array(3)].map(() =>
+          testLogContract.methods.emit_encrypted_events(randomness[++i], preimage[i]).send().wait(),
+        ),
+      );
+      const lastTx = await testLogContract.methods.emit_encrypted_events(randomness[++i], preimage[i]).send().wait();
+
+      const collectedEvent0s = await pxe.getEvents(
```

Comment:
We don't need the `pxe` here no? We could just use the `wallet` instead. Think it would be nicer to do so since it is more likely that devs will be able to call things on the wallet than on the pxe directly.